### ISSUE TITLE
method 'can' on unblessed reference @ Handler.pm

### ISSUE THREAD

**Issue by freddieleeman** [fastmail/authentication_milter#29]
Seeing this in the error log whenever DNS PTR query fails (SERVFAIL):

```
Can't call method "can" on unblessed reference at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication/Handler.pm line 1235.
        ...propagated at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication.pm line 773.

```
`Handler.pm:1235:  if ( $thischild->{'object'}->{'resolver'}->can( 'clear_error_cache' ) ) {`

**Comment by freddieleeman** [comment]
The errors disappeared after increasing the min/max/spare children. Don't know what is causing the errors, doesn't seem to have anything to do with the PTR query, but more the spawning and killing of child processes:

```
Can't call method "can" on unblessed reference at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication/Handler.pm line 1235.
        ...propagated at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication.pm line 773.
Sat Dec  7 16:22:04 2019 authentication_milter[274] Child process 274 starting up
Sat Dec  7 16:22:11 2019 authentication_milter[275] Child process 275 starting up
Sat Dec  7 16:22:29 2019 authentication_milter[265] Child process 265 shutting down
Sat Dec  7 16:22:29 2019 authentication_milter[265] Child exiting
Can't call method "can" on unblessed reference at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication/Handler.pm line 1235.
        ...propagated at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication.pm line 773.
Sat Dec  7 16:22:40 2019 authentication_milter[275] Child process 275 shutting down
Sat Dec  7 16:24:02 2019 authentication_milter[276] Child process 276 starting up
Sat Dec  7 16:24:02 2019 authentication_milter[273] --: ERROR: DNS PTR query failed for 185.234.219.102 with SERVFAIL
Sat Dec  7 16:24:12 2019 authentication_milter[272] Child process 272 shutting down
Sat Dec  7 16:24:12 2019 authentication_milter[272] Child exiting
Can't call method "can" on unblessed reference at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication/Handler.pm line 1235.
        ...propagated at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication.pm line 773.
Sat Dec  7 16:25:57 2019 authentication_milter[277] Child process 277 starting up
Sat Dec  7 16:26:14 2019 authentication_milter[273] Child process 273 shutting down
Sat Dec  7 16:26:14 2019 authentication_milter[273] Child exiting
Can't call method "can" on unblessed reference at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication/Handler.pm line 1235.
        ...propagated at /usr/local/share/perl/5.20.2/Mail/Milter/Authentication.pm line 773.
```

**Comment by marcbradshaw** [comment]
This was fixed in 2.20200106, apologies.
